components/admin_console/user_grid/user_grid.tsx

```diff
@@ -0,0 +1,282 @@
+        const excludeUsersCount = Object.keys(excludeUsers).length;
+        return totalCount + (includeUsersCount - excludeUsersCount);
+    }
+
+    public getPaginationProps = (): {startCount: number; endCount: number; total: number} => {
+        const {includeUsers, excludeUsers, term} = this.props;
+        const {page} = this.state;
+
+        let total: number;
+        let startCount = 0;
+        let endCount = 0;
+
+        if (term === '') {
+            total = this.getVisibleTotalCount();
+            startCount = (page * USERS_PER_PAGE) + 1;
+            endCount = (page + 1) * USERS_PER_PAGE;
+            endCount = endCount > total ? total : endCount;
+        } else {
+            total = this.props.users.length + Object.keys(includeUsers).length;
+            this.props.users.forEach((u) => {
+                if (excludeUsers[u.id]) {
+                    total -= 1;
+                }
+            });
+            endCount = total;
+        }
+        return {startCount, endCount, total};
+    }
+
+    private removeUser = (user: UserProfile) => {
+        const {excludeUsers} = this.props;
+        if (excludeUsers[user.id] === user) {
+            return;
+        }
+
+        let {page} = this.state;
+        const {endCount} = this.getPaginationProps();
+
+        this.props.removeUser(user);
+        if (endCount > this.getVisibleTotalCount() && (endCount % USERS_PER_PAGE) === 1 && page > 0) {
+            page--;
+        }
+
+        this.setState({page});
+    }
+
+    private updateMembership = (membership: BaseMembership) => {
+        const {membershipsToUpdate} = this.state;
+        const {memberships} = this.props;
+        const userId = membership.user_id;
+        membershipsToUpdate[userId] = {
+            ...memberships[userId],
+            ...membership,
+        };
+
+        this.props.updateMembership(membership);
+        this.setState({membershipsToUpdate}, this.forceUpdate);
+    }
+
+    private newMembership = (userId: string): BaseMembership => {
+        return {
+            user_id: userId,
+            scheme_admin: false,
+            scheme_user: true,
+        };
+    }
+
+    private getRows = (): Row[] => {
+        const {page, membershipsToUpdate} = this.state;
+        const {memberships, users, excludeUsers, includeUsers, totalCount, term, scope} = this.props;
+
+        let usersToDisplay = users;
+        const includeUsersList = Object.values(includeUsers);
+
+        // Remove users to remove and add users to add
+        usersToDisplay = usersToDisplay.filter((user) => !excludeUsers[user.id]);
+        usersToDisplay = [...includeUsersList, ...usersToDisplay];
+
+        // Dont load more elements if searching
+        if (term === '') {
+            const {startCount, endCount} = this.getPaginationProps();
+            usersToDisplay = usersToDisplay.slice(startCount - 1, endCount);
+
+            if (usersToDisplay.length < 10 && users.length < totalCount) {
```

Comment:
Could we turn the `10` into a constant by any chance?

Reply:
done